{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8ef4c9ee-1244-4c89-9805-987bffd3bc98",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "repo_dir = '../'\n",
    "\n",
    "seq_length = 2**19*2\n",
    "bin_size = 2048"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7c9be67-b819-497a-8b28-0c6fd7e7a519",
   "metadata": {},
   "source": [
    "# Scoring dnSVs using SuPreMo-Akita"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a5bf30ec-a6c5-467f-b34b-283c484c1f96",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate input for SuPreMo\n",
    "dnSVs = pd.read_csv(f'{repo_dir}data/SFARI_SSC_dnSVs.csv').rename(columns = {'#chrom':'chrom'})\n",
    "dnSVs_for_supremo = dnSVs[[x in ['DEL', 'DUP', 'INV'] for x in dnSVs.svtype]][['chrom', 'pos', 'end', 'svtype', 'svlen']].drop_duplicates().reset_index(drop = True)\n",
    "dnSVs_for_supremo.insert(2, 'REF', '-')\n",
    "dnSVs_for_supremo.insert(3, 'ALT', '-')\n",
    "dnSVs_for_supremo.columns = ['CHROM', 'POS', 'REF', 'ALT', 'END', 'SVTYPE', 'SVLEN']\n",
    "dnSVs_for_supremo.to_csv(f'{repo_dir}variant_scoring/supremo-akita_input/dnSVs_for_SuPreMo.txt', sep = '\\t', index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cdfccfe3-da8b-421f-a934-deef66a248af",
   "metadata": {},
   "source": [
    "Run SuPreMo:\n",
    "\n",
    "python SuPreMo/scripts/SuPreMo.py variant_scoring/supremo-akita_input/dnSVs_for_SuPreMo.txt \\\n",
    "--get_Akita_scores \\\n",
    "--dir variant_scoring/supremo-akita_output \\\n",
    "--file dnSV \\\n",
    "--fa SuPreMo/data/hg38.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "fa3e1b6c-4109-4290-a77b-8984f1ed97e5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>svtype</th>\n",
       "      <th>svlen</th>\n",
       "      <th>MSE</th>\n",
       "      <th>correlation</th>\n",
       "      <th>project</th>\n",
       "      <th>sample</th>\n",
       "      <th>father</th>\n",
       "      <th>father_age_birth_years</th>\n",
       "      <th>mosaic</th>\n",
       "      <th>mother</th>\n",
       "      <th>mother_age_birth_years</th>\n",
       "      <th>role</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6364588</td>\n",
       "      <td>6365496</td>\n",
       "      <td>DEL</td>\n",
       "      <td>908</td>\n",
       "      <td>0.010807</td>\n",
       "      <td>0.975127</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC06730</td>\n",
       "      <td>12707.fa</td>\n",
       "      <td>32.833333</td>\n",
       "      <td>False</td>\n",
       "      <td>12707.mo</td>\n",
       "      <td>31.750000</td>\n",
       "      <td>sib</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8652435</td>\n",
       "      <td>8658053</td>\n",
       "      <td>DEL</td>\n",
       "      <td>5618</td>\n",
       "      <td>0.025499</td>\n",
       "      <td>0.921123</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC08663</td>\n",
       "      <td>13776.fa</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>13776.mo</td>\n",
       "      <td>31.666667</td>\n",
       "      <td>proband</td>\n",
       "      <td>female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10817401</td>\n",
       "      <td>10817860</td>\n",
       "      <td>DEL</td>\n",
       "      <td>459</td>\n",
       "      <td>0.007486</td>\n",
       "      <td>0.988214</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC08564</td>\n",
       "      <td>12834.fa</td>\n",
       "      <td>32.916667</td>\n",
       "      <td>False</td>\n",
       "      <td>12834.mo</td>\n",
       "      <td>29.666667</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>15687875</td>\n",
       "      <td>15696221</td>\n",
       "      <td>DEL</td>\n",
       "      <td>8346</td>\n",
       "      <td>0.012714</td>\n",
       "      <td>0.961685</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC10147</td>\n",
       "      <td>13259.fa</td>\n",
       "      <td>20.666667</td>\n",
       "      <td>False</td>\n",
       "      <td>13259.mo</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>22181932</td>\n",
       "      <td>22182865</td>\n",
       "      <td>DEL</td>\n",
       "      <td>933</td>\n",
       "      <td>0.015066</td>\n",
       "      <td>0.976237</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC12167</td>\n",
       "      <td>14545.fa</td>\n",
       "      <td>34.250000</td>\n",
       "      <td>False</td>\n",
       "      <td>14545.mo</td>\n",
       "      <td>30.500000</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>chrX</td>\n",
       "      <td>122209300</td>\n",
       "      <td>122276000</td>\n",
       "      <td>DEL</td>\n",
       "      <td>66700</td>\n",
       "      <td>0.004793</td>\n",
       "      <td>0.873522</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC12799</td>\n",
       "      <td>14685.fa</td>\n",
       "      <td>30.500000</td>\n",
       "      <td>False</td>\n",
       "      <td>14685.mo</td>\n",
       "      <td>30.083333</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>677</th>\n",
       "      <td>chrX</td>\n",
       "      <td>131051086</td>\n",
       "      <td>131056406</td>\n",
       "      <td>INV</td>\n",
       "      <td>5320</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>0.999905</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC09117</td>\n",
       "      <td>SSC07740</td>\n",
       "      <td>26.083333</td>\n",
       "      <td>False</td>\n",
       "      <td>SSC09122</td>\n",
       "      <td>30.166667</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>chrX</td>\n",
       "      <td>154338501</td>\n",
       "      <td>154589000</td>\n",
       "      <td>DUP</td>\n",
       "      <td>250499</td>\n",
       "      <td>0.108324</td>\n",
       "      <td>0.846746</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC00875</td>\n",
       "      <td>11092.fa</td>\n",
       "      <td>29.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>11092.mo</td>\n",
       "      <td>25.083333</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>679</th>\n",
       "      <td>chrY</td>\n",
       "      <td>5605700</td>\n",
       "      <td>6038500</td>\n",
       "      <td>DUP</td>\n",
       "      <td>432800</td>\n",
       "      <td>0.005075</td>\n",
       "      <td>0.615437</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC12095</td>\n",
       "      <td>14541.fa</td>\n",
       "      <td>33.083333</td>\n",
       "      <td>False</td>\n",
       "      <td>14541.mo</td>\n",
       "      <td>28.750000</td>\n",
       "      <td>proband</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>680</th>\n",
       "      <td>chrY</td>\n",
       "      <td>8357611</td>\n",
       "      <td>8362678</td>\n",
       "      <td>DEL</td>\n",
       "      <td>5067</td>\n",
       "      <td>0.009847</td>\n",
       "      <td>0.511932</td>\n",
       "      <td>SFARI</td>\n",
       "      <td>SSC09163</td>\n",
       "      <td>13818.fa</td>\n",
       "      <td>27.666667</td>\n",
       "      <td>False</td>\n",
       "      <td>13818.mo</td>\n",
       "      <td>26.916667</td>\n",
       "      <td>sib</td>\n",
       "      <td>male</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>681 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom        pos        end svtype   svlen       MSE  correlation project  \\\n",
       "0    chr1    6364588    6365496    DEL     908  0.010807     0.975127   SFARI   \n",
       "1    chr1    8652435    8658053    DEL    5618  0.025499     0.921123   SFARI   \n",
       "2    chr1   10817401   10817860    DEL     459  0.007486     0.988214   SFARI   \n",
       "3    chr1   15687875   15696221    DEL    8346  0.012714     0.961685   SFARI   \n",
       "4    chr1   22181932   22182865    DEL     933  0.015066     0.976237   SFARI   \n",
       "..    ...        ...        ...    ...     ...       ...          ...     ...   \n",
       "676  chrX  122209300  122276000    DEL   66700  0.004793     0.873522   SFARI   \n",
       "677  chrX  131051086  131056406    INV    5320  0.000055     0.999905   SFARI   \n",
       "678  chrX  154338501  154589000    DUP  250499  0.108324     0.846746   SFARI   \n",
       "679  chrY    5605700    6038500    DUP  432800  0.005075     0.615437   SFARI   \n",
       "680  chrY    8357611    8362678    DEL    5067  0.009847     0.511932   SFARI   \n",
       "\n",
       "       sample    father  father_age_birth_years  mosaic    mother  \\\n",
       "0    SSC06730  12707.fa               32.833333   False  12707.mo   \n",
       "1    SSC08663  13776.fa               29.000000   False  13776.mo   \n",
       "2    SSC08564  12834.fa               32.916667   False  12834.mo   \n",
       "3    SSC10147  13259.fa               20.666667   False  13259.mo   \n",
       "4    SSC12167  14545.fa               34.250000   False  14545.mo   \n",
       "..        ...       ...                     ...     ...       ...   \n",
       "676  SSC12799  14685.fa               30.500000   False  14685.mo   \n",
       "677  SSC09117  SSC07740               26.083333   False  SSC09122   \n",
       "678  SSC00875  11092.fa               29.000000   False  11092.mo   \n",
       "679  SSC12095  14541.fa               33.083333   False  14541.mo   \n",
       "680  SSC09163  13818.fa               27.666667   False  13818.mo   \n",
       "\n",
       "     mother_age_birth_years     role     sex  \n",
       "0                 31.750000      sib  female  \n",
       "1                 31.666667  proband  female  \n",
       "2                 29.666667  proband    male  \n",
       "3                 22.000000  proband    male  \n",
       "4                 30.500000  proband    male  \n",
       "..                      ...      ...     ...  \n",
       "676               30.083333  proband    male  \n",
       "677               30.166667  proband    male  \n",
       "678               25.083333  proband    male  \n",
       "679               28.750000  proband    male  \n",
       "680               26.916667      sib    male  \n",
       "\n",
       "[681 rows x 16 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Read SuPreMo output and save to file\n",
    "\n",
    "supremo_scores = pd.concat([dnSVs_for_supremo,\n",
    "                            pd.read_csv(f'{repo_dir}variant_scoring/supremo-akita_output/dnSV_scores', \n",
    "                                        sep = '\\t')],\n",
    "                           axis = 1).drop(['REF', 'ALT', 'var_index'], axis = 1)\n",
    "\n",
    "supremo_scores.columns = ['chrom', 'pos', 'end', 'svtype', 'svlen', 'MSE', 'correlation']\n",
    "\n",
    "dnSVs_scored_noCPX = supremo_scores.merge(dnSVs, on = ['chrom', 'pos', 'end', 'svtype', 'svlen'], how = 'left')\n",
    "\n",
    "dnSVs_scored_noCPX                                      "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63406513-b470-43b3-ba3a-d0c26bcbb9de",
   "metadata": {},
   "source": [
    "## Score CPX variants"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be2e9004-166b-4801-b859-4fbaf669f264",
   "metadata": {},
   "source": [
    "To score CPX variants, the vcf file is need as it contains variant information that is necessary for generating mutated sequences\n",
    "Since that information requires SFARI permissions, we are not sharing it in this repo\n",
    "\n",
    "We use the following variables from the INFO column of the vcf file: CHR2, CPX_TYPE, CPX_INTERVALS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "d58019c8-ea25-4378-ba1e-027795f685e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get previously calculated scores for CPX variants\n",
    "dnSVs_scored_CPX = pd.read_csv(f'{repo_dir}data/dnSVs_scored_CPX.tsv', sep = '\\t')\n",
    "\n",
    "\n",
    "# Combine scores for all dnSVs into one df\n",
    "dnSVs_scored = pd.concat([dnSVs_scored_noCPX, dnSVs_scored_CPX]).reset_index(drop = True)\n",
    "dnSVs_scored['index'] = dnSVs_scored.index\n",
    "\n",
    "dnSVs_scored.to_csv(f'{repo_dir}data/dnSVs_scored', sep = '\\t', index = False)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0b716cf-1f04-4789-bffe-31a6b6820021",
   "metadata": {},
   "outputs": [],
   "source": [
    "# To score CPX, the following funciton was used to get the REF and ALT allele within SuPreMo\n",
    "\n",
    "def get_alleles_CPX(CHR, POS, END, CHR2, CPX_TYPE, CPX_INTERVALS):\n",
    "    \n",
    "    \n",
    "    # Get reference and alternate sequence from REF and ALT allele using reference genome \n",
    "\n",
    "\n",
    "    # Get reference allele\n",
    "    REF_var_seq = fasta_open.fetch(CHR, POS - 1, END - 1).upper()\n",
    "\n",
    "\n",
    "    # Get alternate allele\n",
    "\n",
    "\n",
    "    if CPX_TYPE in ['delINVdel', 'INVdup', 'delINVdup', 'dupINVdup', 'dupINV', 'dupINVdel', \n",
    "                    'delINVdel', 'dupINV', 'dupINVdel', 'dupINVdup', 'INVdup', 'delINVdup']:\n",
    "        # get sequence for inversion\n",
    "\n",
    "        # get sequence to invert\n",
    "        INV_coordinates = CPX_INTERVALS.split('INV_' + CHR2.split('chr')[1] + ':')[1].split(',')[0]\n",
    "        INV_start = int(INV_coordinates.split('-')[0])\n",
    "        INV_end = int(INV_coordinates.split('-')[1])\n",
    "\n",
    "        inv_revcomp = fasta_open.fetch(CHR2, INV_start - 1, INV_end - 1).upper()\n",
    "        inv_seq = str(Seq(inv_revcomp).reverse_complement())\n",
    "\n",
    "\n",
    "\n",
    "        # get sequences left and right of inversion\n",
    "\n",
    "\n",
    "        # Get variant sequence before inversion\n",
    "        if CPX_TYPE in ['delINVdel', 'INVdup', 'delINVdup']:\n",
    "            ALT_left = ''\n",
    "\n",
    "        if CPX_TYPE in ['dupINVdup', 'dupINV', 'dupINVdel']:\n",
    "            DUP1_coordinates = CPX_INTERVALS.split('DUP_' + CHR2.split('chr')[1] + ':')[1].split(',')[0]\n",
    "            DUP1_start = int(DUP1_coordinates.split('-')[0])\n",
    "            DUP1_end = int(DUP1_coordinates.split('-')[1])\n",
    "            ALT_left = fasta_open.fetch(CHR2, DUP1_start - 1, DUP1_end - 1).upper()\n",
    "\n",
    "\n",
    "        # Get variant sequence after inversion\n",
    "        if CPX_TYPE in ['delINVdel', 'dupINV', 'dupINVdel']:\n",
    "            ALT_right = ''\n",
    "\n",
    "        if CPX_TYPE in ['dupINVdup', 'INVdup', 'delINVdup']:\n",
    "\n",
    "            if CPX_TYPE != 'dupINVdup':\n",
    "                num = 1\n",
    "            # if there are 2 duplications, get coordinates for the seconds one\n",
    "            else:\n",
    "                num = 2\n",
    "\n",
    "            DUP2_coordinates = CPX_INTERVALS.split('DUP_' + CHR2.split('chr')[1] + ':')[num].split(',')[0]\n",
    "            DUP2_start = int(DUP2_coordinates.split('-')[0])\n",
    "            DUP2_end = int(DUP2_coordinates.split('-')[1])\n",
    "\n",
    "            ALT_right = fasta_open.fetch(CHR2, DUP2_start - 1, DUP2_end - 1).upper()\n",
    "\n",
    "\n",
    "        # Assemble ALT variant sequence\n",
    "        ALT_var_seq = ALT_left + inv_seq + ALT_right\n",
    "        \n",
    "    else:\n",
    "        raise ValueError('Currently cannot handle complex variants without inversions')\n",
    "\n",
    "\n",
    "    return REF_var_seq, ALT_var_seq\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae20b02d-b525-40d6-9014-60decb446b06",
   "metadata": {},
   "source": [
    "# Scoring dnSVs near CREints using SuPreMo-Akita with weighted scoring"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "050f94a1-8cce-4e10-81fa-ef40cd58f6a0",
   "metadata": {},
   "source": [
    "## Process PLACseq data into CREints for weighting"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc352cc6-b121-40b5-95c0-dc3977b240be",
   "metadata": {},
   "source": [
    "### Get data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c0db284f-0aa5-4c59-a809-8153e80b7398",
   "metadata": {},
   "outputs": [],
   "source": [
    "from pybedtools import BedTool"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3d47330b-c11b-41dd-b24f-bdc7ab127c25",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_49897/4218075283.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  promoter_annot['Start'][promoter_annot['Start'] < 1] = 1\n"
     ]
    }
   ],
   "source": [
    "# Get hg38 promoters of protein-coding genes (Kallisto https://pachterlab.github.io/kallisto/)\n",
    "\n",
    "ensembl = pd.read_csv(f'{repo_dir}data/Homo_sapiens.GRCh38.96.gtf',\n",
    "                     sep='\\t', skiprows=5, header=None, \n",
    "                      names=['chrom','source','desc','start','end','score','strand','score2','desc_further'],\n",
    "                     low_memory=False)\n",
    "ensembl['chr'] = 'chr' + ensembl['chrom'].astype(str)\n",
    "ensembl['name'] = ensembl['desc_further'].str.split('gene_name ').str[1].str.split(';').str[0].str.strip('\"\"')\n",
    "ensembl['biotype'] = ensembl['desc_further'].str.split('biotype ').str[1].str.split(';').str[0].str.strip('\"\"')\n",
    "\n",
    "# Remove nonsense chromosomes\n",
    "chromosomes = []\n",
    "for i in list(range(23))[1:]:\n",
    "    chrom = 'chr' + str(i)\n",
    "    chromosomes.append(chrom)\n",
    "chromosomes.append('chrX')\n",
    "chromosomes.append('chrY')\n",
    "ensembl = ensembl[[x in chromosomes for x in ensembl.chr]]\n",
    "\n",
    "ensembl = ensembl[ensembl.biotype == \"protein_coding\"]\n",
    "ensembl_genes = ensembl.query('desc == \"gene\"').copy()\n",
    "n_ens_genes = pd.DataFrame(ensembl_genes['name'].value_counts())\n",
    "n_ens_genes['gene'] = n_ens_genes.index\n",
    "ensembl_genes_single = n_ens_genes.query('name == 1')['gene'].tolist()\n",
    "ensembl_genes_single_df = ensembl_genes.query('(name in @ensembl_genes_single)').copy()\n",
    "\n",
    "# Get ensemble gene ids\n",
    "ensembl_genes_single_df['gene_id'] = ensembl_genes_single_df['desc_further'].str.split('\"').str[1]\n",
    "ensembl_genes_single_df = ensembl_genes_single_df.rename(columns = {\"name\":\"gene\"})\n",
    "gene_annot = ensembl_genes_single_df[['gene', 'gene_id', 'chr', 'start', 'end', 'strand']]\n",
    "\n",
    "promoter_annot = gene_annot.copy()\n",
    "\n",
    "def label_promoter_start(row):\n",
    "    if row['strand'] == '+' :\n",
    "        return row['start'] - 1000\n",
    "    if row['strand'] == '-' :\n",
    "        return row['end']\n",
    "\n",
    "promoter_annot['Start'] = promoter_annot.apply(lambda row: label_promoter_start(row), axis=1)\n",
    "promoter_annot['Start'][promoter_annot['Start'] < 1] = 1\n",
    "promoter_annot['End'] = promoter_annot['Start'] + 1000\n",
    "\n",
    "promoter_annot_PC_BED = BedTool.from_dataframe(promoter_annot[['chr', 'Start', 'End', 'gene']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d0046878-e2f7-4735-a946-811e0f233c6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get excitatory neuron RNAseq data (Song et al 2021)\n",
    "\n",
    "eN_exp = pd.read_csv(f'{repo_dir}data/eN_avg', sep = '\\t').dropna() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "274a1bb6-f990-44a5-a441-61be7f05f72b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get excitatory neuron ATACseq data (Song et al 2021)\n",
    "\n",
    "ATAC = pd.read_csv(f'{repo_dir}data/eN.ATAC-seq.narrowPeak', sep = '\\t', \n",
    "                   names = ['chrom', 'start', 'end', 'peak_id', 'score', 'strand', 'signalValue', 'pvalue', 'qvalue', 'peak'])\n",
    "ATAC_BED = BedTool.from_dataframe(ATAC[['chrom', 'start', 'end']]).sort()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e7b5430-0b29-4e38-a550-c12fc4115091",
   "metadata": {},
   "source": [
    "### Get loops and filter into CREints"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f0548c26-a1d0-4526-9541-a8f5a2127a26",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get excitatory neuron H3K4me3 PLACseq loops (Song et al 2021)\n",
    "loops = pd.read_csv(f'{repo_dir}data/eN.MAPS.peaks.txt', sep = '\\t').rename(columns = {'chr1':'chrom'})[['chrom', 'start1', 'end1', 'start2', 'end2']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "aabe2645-203c-4658-90c4-aa87b3cc4718",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "97.77350192413414 % of loops are within predictive window\n"
     ]
    }
   ],
   "source": [
    "# Get loops within predictive window \n",
    "\n",
    "dist_cutoff = 900000 #pixel_size*448\n",
    "\n",
    "center_coord1 = [(x+y)/2 for x,y in zip(loops.start1, loops.end1)]\n",
    "center_coord2 = [(x+y)/2 for x,y in zip(loops.start2, loops.end2)]\n",
    "\n",
    "before = len(loops)\n",
    "\n",
    "distance = [abs(x-y) for x,y in zip(center_coord1, center_coord2)]\n",
    "loops = loops[[x < dist_cutoff for x in distance]]\n",
    "\n",
    "print(len(loops)/before*100, '% of loops are within predictive window')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "965cf588-5490-4370-9e2d-36f67005a1ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get left and right loops\n",
    "\n",
    "left_loops = loops[['chrom', 'start1', 'end1']].drop_duplicates().reset_index(drop = True)\n",
    "left_loops['left_index'] = left_loops.index\n",
    "loops = loops.merge(left_loops, on = ['chrom', 'start1', 'end1'])\n",
    "\n",
    "right_loops = loops[['chrom', 'start2', 'end2']].drop_duplicates().reset_index(drop = True)\n",
    "right_loops['right_index'] = right_loops.index\n",
    "loops = loops.merge(right_loops, on = ['chrom', 'start2', 'end2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "c8a935b4-9fec-43d7-94bf-30db8a859a9e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For all loops with the same left anchor, combine right anchors that are within d\n",
    "\n",
    "d = 10000\n",
    "\n",
    "for i in left_loops.left_index:\n",
    "    \n",
    "    loops_i = loops.query('left_index == @i')[['chrom', 'start2', 'end2', 'right_index']]\n",
    "    right_loop_BED = BedTool.from_dataframe(loops_i)\n",
    "\n",
    "    if len(right_loop_BED) > 1:\n",
    "        right_loop_keep = (right_loop_BED\n",
    "                           .sort()\n",
    "                           .merge(d = d, c = 4, o = 'collapse') # collapse indexes of the anchors merged\n",
    "                           .to_dataframe()\n",
    "                           .rename(columns = {'name':'right_index'}))\n",
    "        \n",
    "        if len(right_loop_keep) < len(loops_i):\n",
    "            \n",
    "            right_loop_keep['right_index'] = right_loop_keep['right_index'].str.split(',')\n",
    "            right_loop_keep = right_loop_keep.explode('right_index')\n",
    "\n",
    "            for ii in right_loop_keep.right_index.unique():\n",
    "\n",
    "                loops.loc[(loops.left_index == i) & \n",
    "                          (loops.right_index == int(ii)),\n",
    "                          'start2b'] = int(right_loop_keep.loc[right_loop_keep.right_index == ii,'start'])\n",
    "                loops.loc[(loops.left_index == i) & \n",
    "                          (loops.right_index == int(ii)),\n",
    "                          'end2b'] = int(right_loop_keep.loc[right_loop_keep.right_index == ii,'end'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6756f6a9-6bb2-476e-834f-dcaae310ed5f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat for the right anchor\n",
    "\n",
    "for i in right_loops.right_index:\n",
    "    \n",
    "    loops_i = loops.query('right_index == @i')[['chrom', 'start1', 'end1', 'left_index']]\n",
    "    left_loop_BED = BedTool.from_dataframe(loops_i)\n",
    "\n",
    "    if len(left_loop_BED) > 1:\n",
    "        left_loop_keep = (left_loop_BED\n",
    "                           .sort()\n",
    "                           .merge(d = 10000, c = 4, o = 'collapse')\n",
    "                           .to_dataframe()\n",
    "                           .rename(columns = {'name':'left_index'}))\n",
    "        \n",
    "        if len(left_loop_keep) < len(loops_i):\n",
    "            \n",
    "            left_loop_keep['left_index'] = left_loop_keep['left_index'].str.split(',')\n",
    "            left_loop_keep = left_loop_keep.explode('left_index')\n",
    "\n",
    "            for ii in left_loop_keep.left_index.unique():\n",
    "\n",
    "                loops.loc[(loops.right_index == i) & \n",
    "                          (loops.left_index == int(ii)),\n",
    "                          'start1b'] = int(left_loop_keep.loc[left_loop_keep.left_index == ii,'start'])\n",
    "                loops.loc[(loops.right_index == i) & \n",
    "                          (loops.left_index == int(ii)),\n",
    "                          'end1b'] = int(left_loop_keep.loc[left_loop_keep.left_index == ii,'end'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ac64159e-3f93-410d-a1f3-e122242ee5b9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill in the coordinates of anchors that don't change\n",
    "\n",
    "loops.loc[np.isnan(loops.start1b), \"start1b\"] = loops.loc[np.isnan(loops.start1b), \"start1\"]\n",
    "loops.loc[np.isnan(loops.end1b), \"end1b\"] = loops.loc[np.isnan(loops.end1b), \"end1\"]\n",
    "\n",
    "loops.loc[np.isnan(loops.start2b), \"start2b\"] = loops.loc[np.isnan(loops.start2b), \"start2\"]\n",
    "loops.loc[np.isnan(loops.end2b), \"end2b\"] = loops.loc[np.isnan(loops.end2b), \"end2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "66d5ce06-f38c-44e0-bc1d-6ae908cd5939",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46.76342423390497 % of loops were combined\n"
     ]
    }
   ],
   "source": [
    "# Save the new combined anchors\n",
    "\n",
    "before = len(loops)\n",
    "loops = (loops[['chrom', 'start1b', 'end1b', 'start2b', 'end2b']]\n",
    "         .drop_duplicates()\n",
    "         .rename(columns = {'start1b':'start1', 'end1b':'end1','start2b':'start2', 'end2b':'end2'})\n",
    "         .reset_index(drop = True))\n",
    "\n",
    "loops['Index'] = loops.index\n",
    "\n",
    "print((100 - len(loops)/before*100), \"% of loops were combined\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "55d95517-c89b-4ebf-bae6-ce3ce6b9b6e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get window to look for variants so that the E-P loop and variant can fit in the ~1Mb predictive window\n",
    "\n",
    "dist_cutoff = 900000\n",
    "\n",
    "center_coord1 = [(x+y)/2 for x,y in zip(loops.start1, loops.end1)]\n",
    "center_coord2 = [(x+y)/2 for x,y in zip(loops.start2, loops.end2)]\n",
    "\n",
    "remaining_dist = [dist_cutoff - abs(x-y) for x,y in zip(center_coord1, center_coord2)]\n",
    "\n",
    "# left and right most positions for variant to be in\n",
    "loops['left_cutoff'] = [int(x - y) for x,y in zip(center_coord1, remaining_dist)]\n",
    "loops.loc[loops.left_cutoff < 1,'left_cutoff'] = 1\n",
    "loops['right_cutoff'] = [int(x + y) for x,y in zip(center_coord2, remaining_dist)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "b854d2ab-68a2-4373-a8dd-0e80fc6b521c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "100.0 % of loops have an anchor at a promoter\n"
     ]
    }
   ],
   "source": [
    "# Get loops that overlap a promoter\n",
    "E_or_P = pd.concat(\n",
    "            [loops[['chrom', 'start1', 'end1', 'Index']].rename(columns = {'start1':'start', 'end1':'end'}),\n",
    "            loops[['chrom', 'start2', 'end2', 'Index']].rename(columns = {'start2':'start', 'end2':'end'})],\n",
    "            axis = 0)\n",
    "\n",
    "E_or_P.start = E_or_P.start.astype('int')\n",
    "E_or_P.end = E_or_P.end.astype('int')\n",
    "\n",
    "E_or_P_BED = BedTool.from_dataframe(E_or_P)\n",
    "\n",
    "promoter_loops = (E_or_P_BED\n",
    "                  .intersect(promoter_annot_PC_BED, wa = True, wb = True)\n",
    "                  .to_dataframe()\n",
    "                  .rename(columns = {'name':'loop_index', 'thickEnd':'gene'})\n",
    "                  [['loop_index', 'gene']])\n",
    "\n",
    "before = len(loops)\n",
    "\n",
    "# Only keep promoter loops\n",
    "loops = loops[[x in promoter_loops.loop_index for x in loops.Index]]\n",
    "\n",
    "print(len(loops)/before*100, '% of loops have an anchor at a promoter')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a5b62a5d-df94-47a8-9fc0-2f9f7460fd46",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "69.78465563837295 % of genes are not expressed\n"
     ]
    }
   ],
   "source": [
    "# Get expression data\n",
    "\n",
    "tpm_cutoff = 0.5\n",
    "\n",
    "before = len(eN_exp)\n",
    "\n",
    "# Get only expressed genes\n",
    "eN_exp = eN_exp[eN_exp.TPM >= tpm_cutoff]\n",
    "\n",
    "print((before - len(eN_exp))/before*100, '% of genes are not expressed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "489ecc9b-1b5b-4b65-bff1-fb03054dc0a8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33.031883292626574 % of loops are on promoters that are not expressed\n"
     ]
    }
   ],
   "source": [
    "#Filter only loops that overlap promoters of expressed genes\n",
    "before = len(loops)\n",
    "\n",
    "loops_to_keep = promoter_loops.loop_index[[x in eN_exp.gene.values for x in promoter_loops.gene]].unique()\n",
    "\n",
    "loops = loops[[x in loops_to_keep for x in loops.Index]]\n",
    "\n",
    "print((before - len(loops))/before*100, '% of loops are on promoters that are not expressed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "e33f4380-2555-4fc1-a373-ef1a139f9d94",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find loops near variants\n",
    "loops_windows_BED = BedTool.from_dataframe(loops[['chrom', 'left_cutoff', 'right_cutoff', 'Index']])\n",
    "\n",
    "dnSVs_BED = BedTool.from_dataframe(dnSVs_scored[['chrom', 'pos', 'end', 'index']])\n",
    "\n",
    "loop_var_pairs = (loops_windows_BED\n",
    "                  .intersect(dnSVs_BED, wa = True, wb = True)\n",
    "                  .to_dataframe()\n",
    "                  .rename(columns = {'name':'loop_index', 'thickEnd':'var_index'})\n",
    "                  [['loop_index', 'var_index']])\n",
    "\n",
    "loop_var_pairs = loop_var_pairs.drop_duplicates() # If there are duplicates, they would be removed in the next step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "af344648-84b1-4a4d-8e54-46b307e08e8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "75.80801551389786 % of E-P-variant trios: variant doesnt overlap E/P\n"
     ]
    }
   ],
   "source": [
    "# Get E/Ps that each variant overlaps\n",
    "E_or_P = pd.concat(\n",
    "            [loops[['chrom', 'start1', 'end1', 'Index']].rename(columns = {'start1':'start', 'end1':'end'}),\n",
    "            loops[['chrom', 'start2', 'end2', 'Index']].rename(columns = {'start2':'start', 'end2':'end'})],\n",
    "            axis = 0)\n",
    "E_or_P.start = E_or_P.start.astype('int')\n",
    "E_or_P.end = E_or_P.end.astype('int')\n",
    "\n",
    "E_or_P_BED = BedTool.from_dataframe(E_or_P)\n",
    "\n",
    "var_on_E_or_P = (E_or_P_BED\n",
    "                  .intersect(dnSVs_BED, wa = True, wb = True)\n",
    "                  .to_dataframe()\n",
    "                  .rename(columns = {'name':'loop_index', 'thickEnd':'var_index'})\n",
    "                  [['loop_index', 'var_index']])\n",
    "\n",
    "before = len(loop_var_pairs)\n",
    "\n",
    "# Filter out variants that overlap with E or P of loop they're in a trio with\n",
    "loop_var_pairs = pd.concat([loop_var_pairs, var_on_E_or_P, var_on_E_or_P]).drop_duplicates(keep=False)\n",
    "\n",
    "print(len(loop_var_pairs)/before*100, '% of E-P-variant trios: variant doesnt overlap E/P')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "271387f5-66ec-4242-b75f-0b3465595461",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all loop and var info in one\n",
    "loop_var_coord = loop_var_pairs.copy()\n",
    "\n",
    "# Add info on loops to loop-variant pairs\n",
    "loop_var_coord = (loops\n",
    "                  .rename(columns = {'Index':'loop_index'})\n",
    "                  .drop(['left_cutoff', 'right_cutoff'], axis = 1)\n",
    "                  .merge(loop_var_coord, how = 'right')\n",
    "                 )\n",
    "\n",
    "# Add info on variants to loop-variant pairs\n",
    "loop_var_coord = (dnSVs_scored\n",
    "                  .rename(columns = {'index':'var_index'})\n",
    "                  [['chrom', 'pos', 'end', 'svlen', 'svtype', 'role', 'var_index', 'MSE', 'correlation']]\n",
    "                  .merge(loop_var_coord, how = 'right')\n",
    "                 )\n",
    "\n",
    "loop_var_coord.insert(6, 'center_coord_var', [round((x+y)/2) for x,y in zip(loop_var_coord.end, loop_var_coord.pos)])\n",
    "\n",
    "# Get leftmost and rightmost coordinates of E, P and variant\n",
    "loop_var_coord['start'] = [min(x,y) for x,y in zip(loop_var_coord.start1, \n",
    "                                                    loop_var_coord.pos)]\n",
    "\n",
    "loop_var_coord['end'] = [max(x,y) for x,y in zip(loop_var_coord.end2, \n",
    "                                                    loop_var_coord.end)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b0c835a9-fcc1-486d-8808-f7e0f33db211",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For duplications, center the duplicated sequence not the reference sequence\n",
    "# Extend the variant coordinates on the opposite side of the loop\n",
    "\n",
    "# Annotate variants based on whether they are on the left or right of the loop\n",
    "loop_var_coord.loc[loop_var_coord.pos < loop_var_coord.start1,'var_position'] = 'left'\n",
    "loop_var_coord.loc[loop_var_coord.pos > loop_var_coord.end2,'var_position'] = 'right'\n",
    "\n",
    "# Var to the left of the loop\n",
    "loop_var_coord.loc[(loop_var_coord.svtype == 'DUP') & \n",
    "                   (loop_var_coord.var_position == 'left'),\n",
    "                   'start'] -= loop_var_coord.loc[(loop_var_coord.svtype == 'DUP') & \n",
    "                                                  (loop_var_coord.var_position == 'left'),\n",
    "                                                   'svlen']\n",
    "\n",
    "# Var to the right of the loop\n",
    "loop_var_coord.loc[(loop_var_coord.svtype == 'DUP') & \n",
    "                   (loop_var_coord.var_position == 'right'),\n",
    "                   'end'] += loop_var_coord.loc[(loop_var_coord.svtype == 'DUP') & \n",
    "                                                (loop_var_coord.var_position == 'right'),\n",
    "                                                 'svlen']\n",
    "\n",
    "# Remove trios that exceed prediction window\n",
    "loop_var_coord = loop_var_coord[loop_var_coord.end - loop_var_coord.start < seq_length - (bin_size*64)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1c9a3fc9-13a4-469a-b500-781d29704692",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For duplications, the alternate allele should be able to fit with the loop in the predictive window\n",
    "loop_var_coord.loc[(loop_var_coord.svtype == 'DUP') & \n",
    "                   (loop_var_coord.end - loop_var_coord.start + loop_var_coord.svlen > seq_length-(bin_size*64)),\n",
    "                   'remove'] = 'yes'\n",
    "loop_var_coord = loop_var_coord[loop_var_coord.remove != 'yes'].drop('remove', axis = 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "321c46ae-a48b-4e48-b1b4-ecffc9952723",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50.072150072150066"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# % of variants in predictive window with loop\n",
    "len(loop_var_coord.var_index.unique())/len(dnSVs_scored)*100"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a289c06d-eb4f-4eec-9979-0cae3c292623",
   "metadata": {},
   "source": [
    "### Get variant-specific shift values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "5a0b2182-5472-40c2-8c19-cdc673cfc362",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# Add shift value\n",
    "\n",
    "# when variant is left of the loop\n",
    "loop_var_coord.loc[(loop_var_coord.var_position == 'left'),\n",
    "                   'shift_by'] = [round((loop_end-var_end)/2) for loop_end,var_end in \n",
    "                               zip(loop_var_coord[(loop_var_coord.var_position == 'left')].end2, \n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'left')].end)]\n",
    "\n",
    "# For duplications, consider the variant as the duplicated sequence so it's not out of the window after shifting\n",
    "loop_var_coord.loc[(loop_var_coord.var_position == 'left') & (loop_var_coord.svtype == 'DUP'),\n",
    "                   'shift_by'] = [round((loop_end-var_end)/2+svlen/2) for loop_end,var_end,svlen in \n",
    "                               zip(loop_var_coord[(loop_var_coord.var_position == 'left') & (loop_var_coord.svtype == 'DUP')].end2, \n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'left') & (loop_var_coord.svtype == 'DUP')].end,\n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'left') & (loop_var_coord.svtype == 'DUP')].svlen)]\n",
    "\n",
    "\n",
    "# when variant is right of the loop\n",
    "loop_var_coord.loc[(loop_var_coord.var_position == 'right'),\n",
    "                   'shift_by'] = [-round((var_start-loop_start)/2) for var_start,loop_start in \n",
    "                               zip(loop_var_coord[(loop_var_coord.var_position == 'right')].pos, \n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'right')].start1)]\n",
    "\n",
    "# For duplications, consider the variant as the duplicated sequence so it's not out of the window after shifting\n",
    "loop_var_coord.loc[(loop_var_coord.var_position == 'right') & (loop_var_coord.svtype == 'DUP'),\n",
    "                   'shift_by'] = [-round((var_start-loop_start)/2+svlen/2) for var_start,loop_start,svlen in \n",
    "                               zip(loop_var_coord[(loop_var_coord.var_position == 'right') & (loop_var_coord.svtype == 'DUP')].pos, \n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'right') & (loop_var_coord.svtype == 'DUP')].start1,\n",
    "                                   loop_var_coord[(loop_var_coord.var_position == 'right') & (loop_var_coord.svtype == 'DUP')].svlen)]\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83359ffa-d017-44fb-a485-3aa3507637d7",
   "metadata": {},
   "source": [
    "### Get input and read output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "d2b678b8-73ce-4981-9b59-90ae43f2d8f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get input files for SuPreMo\n",
    "\n",
    "dnSVs_for_supremo_weighted = loop_var_coord[~np.isnan(loop_var_coord.shift_by)][['chrom', 'pos', 'end', 'svtype', 'svlen', 'shift_by', 'var_index']]\n",
    "dnSVs_for_supremo_weighted.insert(2, 'REF', '-')\n",
    "dnSVs_for_supremo_weighted.insert(3, 'ALT', '-')\n",
    "dnSVs_for_supremo_weighted.columns = ['CHROM', 'POS', 'REF', 'ALT', 'END', 'SVTYPE', 'SVLEN', 'shift_by', 'var_index']\n",
    "\n",
    "# input file\n",
    "dnSVs_for_supremo_weighted.drop(['shift_by', 'var_index'], axis = 1).to_csv(f'{repo_dir}variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo.txt', \n",
    "                                           sep = '\\t', index = False)\n",
    "\n",
    "# shift file\n",
    "dnSVs_for_supremo_weighted[['shift_by']].to_csv(f'{repo_dir}variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo_shifts.txt', \n",
    "                                                sep = '\\t', index = False)\n",
    "\n",
    "# Weights file\n",
    "weights_file = pd.concat([loop_var_coord[['chrom', 'start1', 'end1']]\n",
    "                          .rename(columns = {'chrom':'chr', 'start1':'start', 'end1':'end'}),\n",
    "                          loop_var_coord[['chrom', 'start2', 'end2']]\n",
    "                          .rename(columns = {'chrom':'chr', 'start2':'start', 'end2':'end'})],\n",
    "                         axis = 0).reset_index(drop = True)\n",
    "weights_file.to_csv(f'{repo_dir}variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo_weights.txt', \n",
    "                    sep = '\\t', index = False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f26767e1-a5bc-4e94-b894-9ddbe8fb6765",
   "metadata": {},
   "source": [
    "Run SuPreMo:\n",
    "\n",
    "python SuPreMo/scripts/SuPreMo.py variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo.txt \\\n",
    "--get_Akita_scores \\\n",
    "--shifts_file variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo_shifts.txt \\\n",
    "--roi variant_scoring/supremo-akita_input_weighted/EP_for_SuPreMo_weights.txt \\\n",
    "--roi_scales 10 1000000 \\\n",
    "--dir variant_scoring/supremo-akita_output_weighted \\\n",
    "--file EP \\\n",
    "--fa SuPreMo/data/hg38.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "0236d5d0-0e08-4c9f-accf-38327f64c590",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>svlen</th>\n",
       "      <th>svtype</th>\n",
       "      <th>role</th>\n",
       "      <th>var_index</th>\n",
       "      <th>MSE</th>\n",
       "      <th>correlation</th>\n",
       "      <th>start1</th>\n",
       "      <th>end1</th>\n",
       "      <th>start2</th>\n",
       "      <th>end2</th>\n",
       "      <th>loop_index</th>\n",
       "      <th>start</th>\n",
       "      <th>var_position</th>\n",
       "      <th>shift_by</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>212552883</td>\n",
       "      <td>212955000.0</td>\n",
       "      <td>80</td>\n",
       "      <td>DUP</td>\n",
       "      <td>proband</td>\n",
       "      <td>35</td>\n",
       "      <td>0.002183</td>\n",
       "      <td>0.992207</td>\n",
       "      <td>212855000.0</td>\n",
       "      <td>212860000.0</td>\n",
       "      <td>212950000.0</td>\n",
       "      <td>212955000.0</td>\n",
       "      <td>11</td>\n",
       "      <td>212552803.0</td>\n",
       "      <td>left</td>\n",
       "      <td>40.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>70999</td>\n",
       "      <td>DUP</td>\n",
       "      <td>proband</td>\n",
       "      <td>5</td>\n",
       "      <td>0.024572</td>\n",
       "      <td>0.939746</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>27455000.0</td>\n",
       "      <td>27600000.0</td>\n",
       "      <td>27610000.0</td>\n",
       "      <td>13</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-379250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>70999</td>\n",
       "      <td>DUP</td>\n",
       "      <td>proband</td>\n",
       "      <td>5</td>\n",
       "      <td>0.024572</td>\n",
       "      <td>0.939746</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>27455000.0</td>\n",
       "      <td>27600000.0</td>\n",
       "      <td>27630000.0</td>\n",
       "      <td>15</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-379250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>70999</td>\n",
       "      <td>DUP</td>\n",
       "      <td>proband</td>\n",
       "      <td>5</td>\n",
       "      <td>0.024572</td>\n",
       "      <td>0.939746</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>27455000.0</td>\n",
       "      <td>27565000.0</td>\n",
       "      <td>27615000.0</td>\n",
       "      <td>16</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-379250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>70999</td>\n",
       "      <td>DUP</td>\n",
       "      <td>proband</td>\n",
       "      <td>5</td>\n",
       "      <td>0.024572</td>\n",
       "      <td>0.939746</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>27455000.0</td>\n",
       "      <td>27600000.0</td>\n",
       "      <td>27615000.0</td>\n",
       "      <td>17</td>\n",
       "      <td>27285000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-379250.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4686</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>29190000.0</td>\n",
       "      <td>217774</td>\n",
       "      <td>DEL</td>\n",
       "      <td>sib</td>\n",
       "      <td>644</td>\n",
       "      <td>0.065464</td>\n",
       "      <td>0.858959</td>\n",
       "      <td>29070000.0</td>\n",
       "      <td>29075000.0</td>\n",
       "      <td>29185000.0</td>\n",
       "      <td>29190000.0</td>\n",
       "      <td>15142</td>\n",
       "      <td>28295850.0</td>\n",
       "      <td>left</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4687</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28196671</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>17812</td>\n",
       "      <td>DEL</td>\n",
       "      <td>proband</td>\n",
       "      <td>643</td>\n",
       "      <td>0.014642</td>\n",
       "      <td>0.975664</td>\n",
       "      <td>28680000.0</td>\n",
       "      <td>28685000.0</td>\n",
       "      <td>28700000.0</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>15144</td>\n",
       "      <td>28196671.0</td>\n",
       "      <td>left</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4688</th>\n",
       "      <td>chr22</td>\n",
       "      <td>29206135</td>\n",
       "      <td>29234175.0</td>\n",
       "      <td>14020</td>\n",
       "      <td>DUP</td>\n",
       "      <td>sib</td>\n",
       "      <td>645</td>\n",
       "      <td>0.013520</td>\n",
       "      <td>0.965884</td>\n",
       "      <td>28680000.0</td>\n",
       "      <td>28685000.0</td>\n",
       "      <td>28700000.0</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>15144</td>\n",
       "      <td>28680000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-270078.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4689</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>217774</td>\n",
       "      <td>DEL</td>\n",
       "      <td>sib</td>\n",
       "      <td>644</td>\n",
       "      <td>0.065464</td>\n",
       "      <td>0.858959</td>\n",
       "      <td>28680000.0</td>\n",
       "      <td>28685000.0</td>\n",
       "      <td>28700000.0</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>15144</td>\n",
       "      <td>28295850.0</td>\n",
       "      <td>left</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4690</th>\n",
       "      <td>chr22</td>\n",
       "      <td>23641611</td>\n",
       "      <td>23645193.0</td>\n",
       "      <td>1791</td>\n",
       "      <td>DUP</td>\n",
       "      <td>sib</td>\n",
       "      <td>641</td>\n",
       "      <td>0.014930</td>\n",
       "      <td>0.951084</td>\n",
       "      <td>23175000.0</td>\n",
       "      <td>23180000.0</td>\n",
       "      <td>23280000.0</td>\n",
       "      <td>23305000.0</td>\n",
       "      <td>15147</td>\n",
       "      <td>23175000.0</td>\n",
       "      <td>right</td>\n",
       "      <td>-234201.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3186 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom        pos          end   svlen svtype     role  var_index  \\\n",
       "0      chr1  212552883  212955000.0      80    DUP  proband         35   \n",
       "2      chr1   27972501   28114499.0   70999    DUP  proband          5   \n",
       "4      chr1   27972501   28114499.0   70999    DUP  proband          5   \n",
       "5      chr1   27972501   28114499.0   70999    DUP  proband          5   \n",
       "6      chr1   27972501   28114499.0   70999    DUP  proband          5   \n",
       "...     ...        ...          ...     ...    ...      ...        ...   \n",
       "4686  chr22   28295850   29190000.0  217774    DEL      sib        644   \n",
       "4687  chr22   28196671   28720000.0   17812    DEL  proband        643   \n",
       "4688  chr22   29206135   29234175.0   14020    DUP      sib        645   \n",
       "4689  chr22   28295850   28720000.0  217774    DEL      sib        644   \n",
       "4690  chr22   23641611   23645193.0    1791    DUP      sib        641   \n",
       "\n",
       "           MSE  correlation       start1         end1       start2  \\\n",
       "0     0.002183     0.992207  212855000.0  212860000.0  212950000.0   \n",
       "2     0.024572     0.939746   27285000.0   27455000.0   27600000.0   \n",
       "4     0.024572     0.939746   27285000.0   27455000.0   27600000.0   \n",
       "5     0.024572     0.939746   27285000.0   27455000.0   27565000.0   \n",
       "6     0.024572     0.939746   27285000.0   27455000.0   27600000.0   \n",
       "...        ...          ...          ...          ...          ...   \n",
       "4686  0.065464     0.858959   29070000.0   29075000.0   29185000.0   \n",
       "4687  0.014642     0.975664   28680000.0   28685000.0   28700000.0   \n",
       "4688  0.013520     0.965884   28680000.0   28685000.0   28700000.0   \n",
       "4689  0.065464     0.858959   28680000.0   28685000.0   28700000.0   \n",
       "4690  0.014930     0.951084   23175000.0   23180000.0   23280000.0   \n",
       "\n",
       "             end2  loop_index        start var_position  shift_by  \n",
       "0     212955000.0          11  212552803.0         left      40.0  \n",
       "2      27610000.0          13   27285000.0        right -379250.0  \n",
       "4      27630000.0          15   27285000.0        right -379250.0  \n",
       "5      27615000.0          16   27285000.0        right -379250.0  \n",
       "6      27615000.0          17   27285000.0        right -379250.0  \n",
       "...           ...         ...          ...          ...       ...  \n",
       "4686   29190000.0       15142   28295850.0         left       0.0  \n",
       "4687   28720000.0       15144   28196671.0         left       0.0  \n",
       "4688   28720000.0       15144   28680000.0        right -270078.0  \n",
       "4689   28720000.0       15144   28295850.0         left       0.0  \n",
       "4690   23305000.0       15147   23175000.0        right -234201.0  \n",
       "\n",
       "[3186 rows x 17 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loop_var_coord[~np.isnan(loop_var_coord.shift_by)].drop(['center_coord_var', 'start1', 'end1', 'start2', 'end2', 'start', ], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "01026590-590c-430d-a9f8-01bc97c5dbbc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>var_index</th>\n",
       "      <th>mse_HFF_shifted</th>\n",
       "      <th>mse_10-weighted_HFF_shifted</th>\n",
       "      <th>mse_1000000-weighted_HFF_shifted</th>\n",
       "      <th>corr_HFF_shifted</th>\n",
       "      <th>corr_unweighted_HFF_shifted</th>\n",
       "      <th>corr_10-weighted_HFF_shifted</th>\n",
       "      <th>corr_1000000-weighted_HFF_shifted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0.004695</td>\n",
       "      <td>0.004988</td>\n",
       "      <td>0.006108</td>\n",
       "      <td>0.991122</td>\n",
       "      <td>0.011486</td>\n",
       "      <td>0.012042</td>\n",
       "      <td>0.014172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3181</th>\n",
       "      <td>3181</td>\n",
       "      <td>0.016165</td>\n",
       "      <td>0.015632</td>\n",
       "      <td>0.015117</td>\n",
       "      <td>0.956797</td>\n",
       "      <td>0.047436</td>\n",
       "      <td>0.048968</td>\n",
       "      <td>0.050456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3182</th>\n",
       "      <td>3182</td>\n",
       "      <td>0.017810</td>\n",
       "      <td>0.017139</td>\n",
       "      <td>0.015773</td>\n",
       "      <td>0.943461</td>\n",
       "      <td>0.087236</td>\n",
       "      <td>0.071013</td>\n",
       "      <td>0.037968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3183</th>\n",
       "      <td>3183</td>\n",
       "      <td>0.019672</td>\n",
       "      <td>0.019084</td>\n",
       "      <td>0.018457</td>\n",
       "      <td>0.971667</td>\n",
       "      <td>0.038643</td>\n",
       "      <td>0.037469</td>\n",
       "      <td>0.036219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3184</th>\n",
       "      <td>3184</td>\n",
       "      <td>0.031670</td>\n",
       "      <td>0.026324</td>\n",
       "      <td>0.017858</td>\n",
       "      <td>0.936336</td>\n",
       "      <td>0.070892</td>\n",
       "      <td>0.058551</td>\n",
       "      <td>0.039010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3185</th>\n",
       "      <td>3185</td>\n",
       "      <td>0.010071</td>\n",
       "      <td>0.008774</td>\n",
       "      <td>0.006761</td>\n",
       "      <td>0.951447</td>\n",
       "      <td>0.060336</td>\n",
       "      <td>0.048075</td>\n",
       "      <td>0.029045</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3186 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      var_index  mse_HFF_shifted  mse_10-weighted_HFF_shifted  \\\n",
       "0             0         0.004695                     0.004988   \n",
       "1             1         0.015719                     0.018119   \n",
       "2             2         0.015719                     0.018119   \n",
       "3             3         0.015719                     0.018119   \n",
       "4             4         0.015719                     0.018119   \n",
       "...         ...              ...                          ...   \n",
       "3181       3181         0.016165                     0.015632   \n",
       "3182       3182         0.017810                     0.017139   \n",
       "3183       3183         0.019672                     0.019084   \n",
       "3184       3184         0.031670                     0.026324   \n",
       "3185       3185         0.010071                     0.008774   \n",
       "\n",
       "      mse_1000000-weighted_HFF_shifted  corr_HFF_shifted  \\\n",
       "0                             0.006108          0.991122   \n",
       "1                             0.018889          0.958439   \n",
       "2                             0.018889          0.958439   \n",
       "3                             0.018889          0.958439   \n",
       "4                             0.018889          0.958439   \n",
       "...                                ...               ...   \n",
       "3181                          0.015117          0.956797   \n",
       "3182                          0.015773          0.943461   \n",
       "3183                          0.018457          0.971667   \n",
       "3184                          0.017858          0.936336   \n",
       "3185                          0.006761          0.951447   \n",
       "\n",
       "      corr_unweighted_HFF_shifted  corr_10-weighted_HFF_shifted  \\\n",
       "0                        0.011486                      0.012042   \n",
       "1                        0.045866                      0.037107   \n",
       "2                        0.045866                      0.037107   \n",
       "3                        0.045866                      0.037107   \n",
       "4                        0.045866                      0.037107   \n",
       "...                           ...                           ...   \n",
       "3181                     0.047436                      0.048968   \n",
       "3182                     0.087236                      0.071013   \n",
       "3183                     0.038643                      0.037469   \n",
       "3184                     0.070892                      0.058551   \n",
       "3185                     0.060336                      0.048075   \n",
       "\n",
       "      corr_1000000-weighted_HFF_shifted  \n",
       "0                              0.014172  \n",
       "1                              0.034296  \n",
       "2                              0.034296  \n",
       "3                              0.034296  \n",
       "4                              0.034296  \n",
       "...                                 ...  \n",
       "3181                           0.050456  \n",
       "3182                           0.037968  \n",
       "3183                           0.036219  \n",
       "3184                           0.039010  \n",
       "3185                           0.029045  \n",
       "\n",
       "[3186 rows x 8 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(f'{repo_dir}variant_scoring/supremo-akita_output_weighted/EP_scores', \n",
    "                                        sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "72f5fe75-b5e8-4501-908c-5235b53e5b93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chrom', 'pos', 'end', 'svlen', 'svtype', 'role', 'center_coord_var',\n",
       "       'var_index', 'MSE', 'correlation', 'start1', 'end1', 'start2', 'end2',\n",
       "       'loop_index', 'start', 'var_position', 'shift_by', 'var_index',\n",
       "       'mse_HFF_shifted', 'mse_10-weighted_HFF_shifted',\n",
       "       'mse_1000000-weighted_HFF_shifted', 'corr_HFF_shifted',\n",
       "       'corr_unweighted_HFF_shifted', 'corr_10-weighted_HFF_shifted',\n",
       "       'corr_1000000-weighted_HFF_shifted'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.concat([loop_var_coord[~np.isnan(loop_var_coord.shift_by)],\n",
    "                                    pd.read_csv(f'{repo_dir}variant_scoring/supremo-akita_output_weighted/EP_scores', \n",
    "                                                sep = '\\t')],\n",
    "                                   axis = 1).columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "730fbe1f-489f-4685-b031-c0f394055125",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>END</th>\n",
       "      <th>SVTYPE</th>\n",
       "      <th>SVLEN</th>\n",
       "      <th>shift_by</th>\n",
       "      <th>var_index</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>212552883</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>212955000.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>80</td>\n",
       "      <td>40.0</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4686</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>29190000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>217774</td>\n",
       "      <td>0.0</td>\n",
       "      <td>644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4687</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28196671</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>17812</td>\n",
       "      <td>0.0</td>\n",
       "      <td>643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4688</th>\n",
       "      <td>chr22</td>\n",
       "      <td>29206135</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>29234175.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>14020</td>\n",
       "      <td>-270078.0</td>\n",
       "      <td>645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4689</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>217774</td>\n",
       "      <td>0.0</td>\n",
       "      <td>644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4690</th>\n",
       "      <td>chr22</td>\n",
       "      <td>23641611</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>23645193.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>1791</td>\n",
       "      <td>-234201.0</td>\n",
       "      <td>641</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3186 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      CHROM        POS REF ALT          END SVTYPE   SVLEN  shift_by  \\\n",
       "0      chr1  212552883   -   -  212955000.0    DUP      80      40.0   \n",
       "2      chr1   27972501   -   -   28114499.0    DUP   70999 -379250.0   \n",
       "4      chr1   27972501   -   -   28114499.0    DUP   70999 -379250.0   \n",
       "5      chr1   27972501   -   -   28114499.0    DUP   70999 -379250.0   \n",
       "6      chr1   27972501   -   -   28114499.0    DUP   70999 -379250.0   \n",
       "...     ...        ...  ..  ..          ...    ...     ...       ...   \n",
       "4686  chr22   28295850   -   -   29190000.0    DEL  217774       0.0   \n",
       "4687  chr22   28196671   -   -   28720000.0    DEL   17812       0.0   \n",
       "4688  chr22   29206135   -   -   29234175.0    DUP   14020 -270078.0   \n",
       "4689  chr22   28295850   -   -   28720000.0    DEL  217774       0.0   \n",
       "4690  chr22   23641611   -   -   23645193.0    DUP    1791 -234201.0   \n",
       "\n",
       "      var_index  \n",
       "0            35  \n",
       "2             5  \n",
       "4             5  \n",
       "5             5  \n",
       "6             5  \n",
       "...         ...  \n",
       "4686        644  \n",
       "4687        643  \n",
       "4688        645  \n",
       "4689        644  \n",
       "4690        641  \n",
       "\n",
       "[3186 rows x 9 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dnSVs_for_supremo_weighted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "bc27a671-d142-4ec1-8f7a-273215c78ea3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read SuPreMo output and save to file\n",
    "\n",
    "dnSVs_scored_weighted = pd.concat([dnSVs_for_supremo_weighted.reset_index(drop = True),\n",
    "                                   pd.read_csv(f'{repo_dir}variant_scoring/supremo-akita_output_weighted/EP_scores', \n",
    "                                               sep = '\\t').drop('var_index', axis = 1)], \n",
    "                                  axis = 1).drop(['REF', 'ALT'], axis = 1)\n",
    "\n",
    "dnSVs_scored_weighted.columns = ['chrom', 'pos', 'end', 'svtype', 'svlen', 'shift_by', 'var_index',\n",
    "                                 'MSE', 'MSE_weighted', 'MSE_CREint',\n",
    "                                 'correlation', 'correlation_unweighted',\n",
    "                                 'correlation_weighted', 'correlation_CREint']\n",
    "\n",
    "dnSVs_scored_weighted.to_csv(f'{repo_dir}data/dnSVs_scored_weighted', sep = '\\t', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
