{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a0d0c4c7-be3b-40cf-86c2-7e2d6fb5fb32",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pybedtools import BedTool\n",
    "\n",
    "repo_dir = '../'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f0354e67-5fb2-4c30-827c-b5265808f4eb",
   "metadata": {},
   "source": [
    "# Get data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e8f9005a-31b8-48de-9e21-58407ecb4c0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get variants\n",
    "\n",
    "# all variants\n",
    "SSC_dnSV_all = pd.read_csv(f'{repo_dir}data/SFARI_SSC_dnSVs.csv').rename(columns = {'#chrom':'chrom'})\n",
    "SSC_dnSV_all['index_all'] = SSC_dnSV_all.index\n",
    "\n",
    "# scored variants\n",
    "SSC_dnSV_scored = pd.read_csv(f'{repo_dir}data/dnSVs_scored', sep = '\\t', low_memory=False)\n",
    "\n",
    "indexes = SSC_dnSV_all.merge(SSC_dnSV_scored, how = 'left')[['index', 'index_all']]\n",
    "\n",
    "# Add both indexes to both dfs\n",
    "SSC_dnSV_all = SSC_dnSV_all.merge(indexes, how = 'left')\n",
    "SSC_dnSV_scored = SSC_dnSV_scored.merge(indexes, how = 'left')\n",
    "\n",
    "# Get BED files\n",
    "SSC_dnSV_all_BED = BedTool.from_dataframe(SSC_dnSV_all[['chrom', 'pos', 'end', 'index', 'index_all']])\n",
    "SSC_dnSV_scored_BED = BedTool.from_dataframe(SSC_dnSV_scored[['chrom', 'pos', 'end', 'index', 'index_all']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "5fdc2f14-69f7-4ff7-b5d8-eeb15ded06b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Variant scores\n",
    "scores = SSC_dnSV_scored[~np.isnan(SSC_dnSV_scored.correlation)]\n",
    "weighted_scores = pd.read_csv(f'{repo_dir}data/dnSVs_scored_weighted', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "643c437e-5daf-4cfc-8eff-9eb5a843e858",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_47747/1277924221.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  SV_window.pos = SV_window.pos - dist\n",
      "/tmp/ipykernel_47747/1277924221.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  SV_window.end = SV_window.end + dist\n"
     ]
    }
   ],
   "source": [
    "# Get window around variants\n",
    "dist = 500000\n",
    "\n",
    "SV_window = SSC_dnSV_scored[['chrom', 'pos', 'end', 'index']]\n",
    "SV_window.pos = SV_window.pos - dist\n",
    "SV_window.loc[SV_window.pos < 0, \"pos\"] = 0\n",
    "SV_window.end = SV_window.end + dist\n",
    "SV_window_BED = BedTool.from_dataframe(SV_window)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7c61d4e9-791a-454e-bb65-eeb7ae460136",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get exon annotations\n",
    "gencode_annotations = pd.read_csv(f'{repo_dir}data/gencode.v46.annotation.gtf.gz', \n",
    "                                  sep='\\t', skiprows=5, header=None, \n",
    "                                  names=['chrom','source','desc','start','end','score','strand','score2','desc_further'],\n",
    "                                  low_memory=False)\n",
    "gencode_annotations['gene'] = gencode_annotations['desc_further'].str.split('gene_name ').str[1].str.split(';').str[0].str.strip('\"\"')\n",
    "gencode_annotations['annotation'] = gencode_annotations.gene + '_' + gencode_annotations.desc\n",
    "\n",
    "gene_exons = gencode_annotations[['exon' in x for x in gencode_annotations.annotation]]\n",
    "\n",
    "gene_exons_BED = BedTool.from_dataframe(gene_exons[['chrom', 'start', 'end', 'gene']])\n",
    "\n",
    "\n",
    "# Get variants that fall on exons\n",
    "vars_on_exons = (SSC_dnSV_all_BED\n",
    "                 .intersect(gene_exons_BED, wa = True, wb = True)\n",
    "                 .to_dataframe()\n",
    "                .rename(columns = {'name':'index', 'score':'index_all', 'itemRgb':'Gene'})\n",
    "                [['index', 'index_all', 'Gene']]\n",
    "                .drop_duplicates())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1db1ed6f-3ddf-4eb4-a3d3-f33a1b9240cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get ASD genes (SFARI, Satterstrom, and An)\n",
    "ASD_genes = pd.read_csv(f'{repo_dir}data/ASD_genes', sep = '\\t').Gene.values\n",
    "\n",
    "# All gene coordinates\n",
    "gene_coords = pd.read_csv('/pollard/data/projects/kgjoni/genome/hg38/gene_annot_PC')[['chr', 'start', 'end', 'gene']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "51d47a1a-7068-4d17-a18e-8f4b06c163ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "ASD_dir = '/pollard/data/projects/kgjoni/Akita/ASD_project/'\n",
    "\n",
    "# RNAseq\n",
    "eN_exp = pd.read_csv(f'{ASD_dir}Song2020/shen_bdbag_2019/data/embargo/shen/transcriptome/processed/counts/eN_avg', sep = '\\t').dropna() \n",
    "\n",
    "\n",
    "# Excitatory neuron regulatory regions\n",
    "\n",
    "# Poised promoters\n",
    "H3K27me3 = pd.read_csv(f'{ASD_dir}GSE177154_ENCFF154YOB_pseudoreplicated_peaks_GRCh38.bed', sep = '\\t',\n",
    "                       names = ['chrom', 'start', 'end', 'peak_id', 'score', 'strand', 'signalValue', 'pvalue', 'qvalue', 'peak'])\n",
    "# Poised enhancers\n",
    "H3K4me1 = pd.read_csv(f'{ASD_dir}ENCFF904VQK.bed', sep = '\\t',\n",
    "                       names = ['chrom', 'start', 'end', 'peak_id', 'score', 'strand', 'signalValue', 'pvalue', 'qvalue', 'peak'])\n",
    "# Active enhancers\n",
    "H3K27Ac = pd.read_csv(f'{ASD_dir}ENCFF857GOV.bed', sep = '\\t',\n",
    "                       names = ['chrom', 'start', 'end', 'peak_id', 'score', 'strand', 'signalValue', 'pvalue', 'qvalue', 'peak'])\n",
    "\n",
    "# All regulatory_marks\n",
    "eN_reg_marks = pd.concat([H3K27me3[['chrom', 'start', 'end']],\n",
    "                          H3K4me1[['chrom', 'start', 'end']],\n",
    "                          H3K27Ac[['chrom', 'start', 'end']]], axis = 0)\n",
    "\n",
    "eN_reg_marks_BED = BedTool.from_dataframe(eN_reg_marks).sort().merge()\n",
    "\n",
    "# CTCF\n",
    "CTCF = pd.read_csv(f'{ASD_dir}ENCFF579WDY.bed', sep = '\\t',\n",
    "                       names = ['chrom', 'start', 'end', 'peak_id', 'score', 'strand', 'signalValue', 'pvalue', 'qvalue', 'peak'])\n",
    "CTCF_BED = BedTool.from_dataframe(CTCF[['chrom', 'start', 'end']]).sort()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3ce6525-cd6c-4327-a72a-34bbcca867c5",
   "metadata": {},
   "source": [
    "# Get variants that pass each criteria"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "076c816e-1672-461a-acbd-18d7dfd2694b",
   "metadata": {},
   "source": [
    "We used the following criteria for prioritizing candidate dnSVs.\n",
    "\n",
    "Required:\n",
    "\n",
    "1. Not on ASD gene: Variant does not overlap ASD gene exon.\n",
    "2. No causal dnSV: Proband does not have dnSV overlapping ASD gene exon.\n",
    "3. Good prediction: Predicted maps for the reference genome around dnSV are similar to NPC experimental HiC maps. This includes dnSVs for which the mean squared error (MSE) between the Akita predicted and the experimental contact maps is less than the 85th percentile of MSE across all scored variants.\n",
    "4. No sibling dnSV: There are no similar variants in siblings. This includes proband dnSVs that don’t overlap by more than half of the variant region with more than half of any sibling dnSV region.\n",
    "5. Disrupts CREint: Variant results in changed contact at CREints. This includes dnSVs with weighted disruption scores at CREints (Fig. 2Biv) larger than the 65th percentile across all standard scores (Fig. 1B).\n",
    "6. Near ASD gene: Variant is within 500 kb of an ASD gene.\n",
    "7. Disruptive: Variant is disruptive to 3D genome folding of surrounding regions. This includes dnSVs with disruption scores above the 65th percentile across all scores.\n",
    "\n",
    "Optional:\n",
    "\n",
    "8. Not on expressed gene: Variant does not overlap expressed gene (TPM > 0.5) exon.\n",
    "9. Not on ExN RE: Variant does not overlap ExN regulatory elements, namely active enhancers (H3K27Ac), poised enhancers (H3K4me1), and poised promoters (H3K27me3) as defined by their respective ChIP-Seq peaks. \n",
    "10. Change on CREint: Variant disruption focused on CREints, meaning weighted score (Fig. 2Biii) is higher than unweighted score (Fig. 2Bii).\n",
    "11. On CTCF: Variant overlaps less than half of any ExN CTCF ChIP-Seq peak.\n",
    "12. Deletion: Variant is a deletion. These are the most straightforward to edit in cells."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "db26a256-168a-4b52-89b2-f8f90b120c94",
   "metadata": {},
   "outputs": [],
   "source": [
    "variant_criteria = SSC_dnSV_scored[['index']]\n",
    "var_list = {}\n",
    "criteria = ['disruptive', 'not_ASD_gene', 'no_causal_var', 'good_pred', 'no_sib_var', \n",
    "            'disrupt_CREint', 'not_expr_gene', 'not_ExN_RE', 'near_ASD_gene', 'change_on_CREint', 'on_CTCF', 'is_deletion']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c290f221-875f-410d-ba3c-757455f26d8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(209, 598, 34.94983277591973)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 1: Variant is disruptive to 3D genome folding of surrounding regions\n",
    "\n",
    "# Get percentile\n",
    "# from scipy import stats\n",
    "# stats.percentileofscore(1 - SSC_results.correlation, \n",
    "#                         min(1 - SSC_results[[x in selected_vars for x in SSC_results['index']]].correlation), \n",
    "#                         kind='rank')\n",
    "\n",
    "# Get SVs with disruption scores in the top half\n",
    "disruption_cutoff = np.nanquantile(1 - scores.correlation, 0.65)\n",
    "var_list['disruptive'] = scores[1 - scores.correlation > disruption_cutoff]['index'].values\n",
    "\n",
    "len(var_list['disruptive']), len(scores), len(var_list['disruptive'])/len(scores)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8201c843-f613-4e18-8b67-2e73a7428d47",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(320, 598, 53.51170568561873)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 2: Variant is near known ASD genes\n",
    "\n",
    "ASD_gene_coords_BED = BedTool.from_dataframe(gene_coords[[x in ASD_genes for x in gene_coords.gene]])\n",
    "\n",
    "var_list['near_ASD_gene'] = np.unique(SV_window_BED\n",
    "                               .intersect(ASD_gene_coords_BED, wa = True, wb = True)\n",
    "                               .to_dataframe()\n",
    "                               .name\n",
    "                               .values)\n",
    "\n",
    "len(var_list['near_ASD_gene']), len(scores), len(var_list['near_ASD_gene'])/len(scores)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a62c07d4-fc30-4658-b02c-c87b2c839012",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_47747/2374146237.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  vars_on_exons_['index'] = vars_on_exons_['index'].astype('float').astype('int')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(563, 598, 94.14715719063545)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 3: Variant does not overlap exon of ASD-associated gene\n",
    "\n",
    "# Get variants that don't overlap exons of ASD genes\n",
    "vars_on_exons_ = vars_on_exons[vars_on_exons['index'] != '.']\n",
    "vars_on_exons_['index'] = vars_on_exons_['index'].astype('float').astype('int')\n",
    "exclude = np.unique(vars_on_exons_[[x in ASD_genes for x in vars_on_exons_.Gene]]['index'].values)\n",
    "var_list['not_ASD_gene'] = variant_criteria[[x not in exclude for x in variant_criteria['index']]]['index'].values\n",
    "\n",
    "len(var_list['not_ASD_gene']), len(scores), len(var_list['not_ASD_gene'])/len(scores)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "6c4d2096-29ca-4a45-be4f-f19b1dcd1a9d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(541, 693, 78.06637806637806)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 4: Individual does not have exonic de novo variant on ASD gene\n",
    "\n",
    "# Get individuals with vars on ASD exons\n",
    "individuals_to_exclude = (vars_on_exons\n",
    "                          [[x in ASD_genes for x in vars_on_exons.Gene]]\n",
    "                          [['index_all', 'Gene']]\n",
    "                          .merge(SSC_dnSV_all[['index_all', 'index', 'sample']], \n",
    "                                 how = 'left')\n",
    "                          ['sample']\n",
    "                          .values)\n",
    "\n",
    "# Get variants from individuals without ASD exonic dn vars\n",
    "var_list['no_causal_var'] = SSC_dnSV_scored[[x not in individuals_to_exclude for x in SSC_dnSV_scored['sample']]]['index'].values\n",
    "\n",
    "len(var_list['no_causal_var']), len(variant_criteria), len(var_list['no_causal_var'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "b93f91a3-a86c-4d59-9d45-7c97cd6194b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(521, 693, 75.18037518037518)"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 5: Predicted maps for the reference look similar to NPC experimental HiC maps\n",
    "\n",
    "predicted_vs_experimental_scores = pd.read_table(f'{repo_dir}data/predicted_vs_experimental_scores', sep = '\\t')\n",
    "\n",
    "pred_cutoff = np.nanquantile(predicted_vs_experimental_scores.MSE_HFF_MicroC, 0.85)\n",
    "\n",
    "var_list['good_pred'] = predicted_vs_experimental_scores[predicted_vs_experimental_scores.MSE_HFF_MicroC < pred_cutoff]['index'].values\n",
    "\n",
    "len(var_list['good_pred']), len(variant_criteria), len(var_list['good_pred'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "5986cac5-7c2b-485c-880b-a078b59f59dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(411, 693, 59.307359307359306)"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 6: There are no similar variants with similar disruption in siblings\n",
    "\n",
    "# Proband variants that don't overlap a sibling variant by more than half of each (include sibling not scored variants)\n",
    "exclude = (BedTool.from_dataframe(SSC_dnSV_scored\n",
    "                                  [SSC_dnSV_scored.role == 'proband']\n",
    "                                  [['chrom', 'pos', 'end', 'index']])\n",
    "           .intersect(BedTool.from_dataframe(SSC_dnSV_all\n",
    "                                             [SSC_dnSV_all.role == 'sib']\n",
    "                                             [['chrom', 'pos', 'end']]), f = 0.5, r = True)\n",
    "           .to_dataframe()\n",
    "           .name\n",
    "           .values)\n",
    "\n",
    "var_list['no_sib_var'] = SSC_dnSV_scored[(SSC_dnSV_scored.role == 'proband') & [x not in exclude for x in SSC_dnSV_scored['index']]]['index'].values\n",
    "\n",
    "len(var_list['no_sib_var']), len(variant_criteria), len(var_list['no_sib_var'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "743a5c27-370a-4f73-8ea0-e93687ff4ff2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>end</th>\n",
       "      <th>svtype</th>\n",
       "      <th>svlen</th>\n",
       "      <th>shift_by</th>\n",
       "      <th>MSE</th>\n",
       "      <th>MSE_weighted</th>\n",
       "      <th>MSE_CREint</th>\n",
       "      <th>correlation</th>\n",
       "      <th>correlation_unweighted</th>\n",
       "      <th>correlation_weighted</th>\n",
       "      <th>correlation_CREint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>212552883</td>\n",
       "      <td>212955000.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>80</td>\n",
       "      <td>40.0</td>\n",
       "      <td>0.004695</td>\n",
       "      <td>0.004988</td>\n",
       "      <td>0.006108</td>\n",
       "      <td>0.991122</td>\n",
       "      <td>0.011486</td>\n",
       "      <td>0.012042</td>\n",
       "      <td>0.014172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27972501</td>\n",
       "      <td>28114499.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>70999</td>\n",
       "      <td>-379250.0</td>\n",
       "      <td>0.015719</td>\n",
       "      <td>0.018119</td>\n",
       "      <td>0.018889</td>\n",
       "      <td>0.958439</td>\n",
       "      <td>0.045866</td>\n",
       "      <td>0.037107</td>\n",
       "      <td>0.034296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3181</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>29190000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>217774</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.016165</td>\n",
       "      <td>0.015632</td>\n",
       "      <td>0.015117</td>\n",
       "      <td>0.956797</td>\n",
       "      <td>0.047436</td>\n",
       "      <td>0.048968</td>\n",
       "      <td>0.050456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3182</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28196671</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>17812</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.017810</td>\n",
       "      <td>0.017139</td>\n",
       "      <td>0.015773</td>\n",
       "      <td>0.943461</td>\n",
       "      <td>0.087236</td>\n",
       "      <td>0.071013</td>\n",
       "      <td>0.037968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3183</th>\n",
       "      <td>chr22</td>\n",
       "      <td>29206135</td>\n",
       "      <td>29234175.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>14020</td>\n",
       "      <td>-270078.0</td>\n",
       "      <td>0.019672</td>\n",
       "      <td>0.019084</td>\n",
       "      <td>0.018457</td>\n",
       "      <td>0.971667</td>\n",
       "      <td>0.038643</td>\n",
       "      <td>0.037469</td>\n",
       "      <td>0.036219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3184</th>\n",
       "      <td>chr22</td>\n",
       "      <td>28295850</td>\n",
       "      <td>28720000.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>217774</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.031670</td>\n",
       "      <td>0.026324</td>\n",
       "      <td>0.017858</td>\n",
       "      <td>0.936336</td>\n",
       "      <td>0.070892</td>\n",
       "      <td>0.058551</td>\n",
       "      <td>0.039010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3185</th>\n",
       "      <td>chr22</td>\n",
       "      <td>23641611</td>\n",
       "      <td>23645193.0</td>\n",
       "      <td>DUP</td>\n",
       "      <td>1791</td>\n",
       "      <td>-234201.0</td>\n",
       "      <td>0.010071</td>\n",
       "      <td>0.008774</td>\n",
       "      <td>0.006761</td>\n",
       "      <td>0.951447</td>\n",
       "      <td>0.060336</td>\n",
       "      <td>0.048075</td>\n",
       "      <td>0.029045</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3186 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom        pos          end svtype   svlen  shift_by       MSE  \\\n",
       "0      chr1  212552883  212955000.0    DUP      80      40.0  0.004695   \n",
       "1      chr1   27972501   28114499.0    DUP   70999 -379250.0  0.015719   \n",
       "2      chr1   27972501   28114499.0    DUP   70999 -379250.0  0.015719   \n",
       "3      chr1   27972501   28114499.0    DUP   70999 -379250.0  0.015719   \n",
       "4      chr1   27972501   28114499.0    DUP   70999 -379250.0  0.015719   \n",
       "...     ...        ...          ...    ...     ...       ...       ...   \n",
       "3181  chr22   28295850   29190000.0    DEL  217774       0.0  0.016165   \n",
       "3182  chr22   28196671   28720000.0    DEL   17812       0.0  0.017810   \n",
       "3183  chr22   29206135   29234175.0    DUP   14020 -270078.0  0.019672   \n",
       "3184  chr22   28295850   28720000.0    DEL  217774       0.0  0.031670   \n",
       "3185  chr22   23641611   23645193.0    DUP    1791 -234201.0  0.010071   \n",
       "\n",
       "      MSE_weighted  MSE_CREint  correlation  correlation_unweighted  \\\n",
       "0         0.004988    0.006108     0.991122                0.011486   \n",
       "1         0.018119    0.018889     0.958439                0.045866   \n",
       "2         0.018119    0.018889     0.958439                0.045866   \n",
       "3         0.018119    0.018889     0.958439                0.045866   \n",
       "4         0.018119    0.018889     0.958439                0.045866   \n",
       "...            ...         ...          ...                     ...   \n",
       "3181      0.015632    0.015117     0.956797                0.047436   \n",
       "3182      0.017139    0.015773     0.943461                0.087236   \n",
       "3183      0.019084    0.018457     0.971667                0.038643   \n",
       "3184      0.026324    0.017858     0.936336                0.070892   \n",
       "3185      0.008774    0.006761     0.951447                0.060336   \n",
       "\n",
       "      correlation_weighted  correlation_CREint  \n",
       "0                 0.012042            0.014172  \n",
       "1                 0.037107            0.034296  \n",
       "2                 0.037107            0.034296  \n",
       "3                 0.037107            0.034296  \n",
       "4                 0.037107            0.034296  \n",
       "...                    ...                 ...  \n",
       "3181              0.048968            0.050456  \n",
       "3182              0.071013            0.037968  \n",
       "3183              0.037469            0.036219  \n",
       "3184              0.058551            0.039010  \n",
       "3185              0.048075            0.029045  \n",
       "\n",
       "[3186 rows x 13 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weighted_scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "f822fad3-ef08-419b-80d2-6a8b4ca7861a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(312, 693, 45.02164502164502)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 7: Variant results in changed contact between CRE pair(s)\n",
    "\n",
    "var_list['disrupt_CREint'] = np.unique(weighted_scores[1 - weighted_scores['correlation_CREint'] > disruption_cutoff]['var_index'].values)\n",
    "\n",
    "len(var_list['disrupt_CREint']), len(variant_criteria), len(var_list['disrupt_CREint'])/len(variant_criteria)*100\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "6f37f602-b377-448b-aefd-433710471ff7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_47747/2221596772.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  vars_on_exons_['index'] = vars_on_exons_['index'].astype('float').astype('int')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(440, 693, 63.49206349206349)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 8: Variant does not overlap exon of expressed genes\n",
    "\n",
    "# Get variants that don't overlap exons or promoters of expressed genes\n",
    "expressed_genes = eN_exp[eN_exp.TPM > 0.5].gene.values\n",
    "vars_on_exons_ = vars_on_exons[vars_on_exons['index'] != '.']\n",
    "vars_on_exons_['index'] = vars_on_exons_['index'].astype('float').astype('int')\n",
    "exclude = np.unique(vars_on_exons_[[x in expressed_genes for x in vars_on_exons_.Gene]]['index'].values)\n",
    "\n",
    "var_list['not_expr_gene'] = variant_criteria[[x not in exclude for x in variant_criteria['index']]]['index'].values\n",
    "\n",
    "len(var_list['not_expr_gene']), len(variant_criteria), len(var_list['not_expr_gene'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "9bd9446c-4ba1-45aa-a8de-60210d5c7e66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(330, 693, 47.61904761904761)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 9: Variant does not overlap excitatory neuron regulatory regions\n",
    "\n",
    "exclude = np.unique(SSC_dnSV_scored_BED.intersect(eN_reg_marks_BED).to_dataframe().name.values)\n",
    "var_list['not_ExN_RE'] = variant_criteria[[x not in exclude for x in variant_criteria['index']]]['index'].values\n",
    "\n",
    "len(var_list['not_ExN_RE']), len(variant_criteria), len(var_list['not_ExN_RE'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "2b5c809e-22e2-4f63-bbbb-5cd975dc387f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(205, 693, 29.58152958152958)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 10: Variant disruption score up-weighting CREints is higher than unweighted score\n",
    "\n",
    "var_list['change_on_CREint'] = np.unique(weighted_scores[[x>y for x,y in zip(weighted_scores['correlation_weighted'], \n",
    "                                                                             weighted_scores['correlation_unweighted'])]]['var_index'].values)\n",
    "\n",
    "len(var_list['change_on_CREint']), len(variant_criteria), len(var_list['change_on_CREint'])/len(variant_criteria)*100\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "d419e82d-3431-4c72-b42a-39847d2b4067",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(201, 693, 29.004329004329005)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 11: Variant overlaps CTCF binding site by at least half, on_CTCF\n",
    "\n",
    "var_list['on_CTCF'] = np.unique(SSC_dnSV_scored_BED.intersect(CTCF_BED, F = 0.5).to_dataframe().name.values)\n",
    "\n",
    "len(var_list['on_CTCF']), len(variant_criteria), len(var_list['on_CTCF'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "dcdc69a2-625d-4a00-9a70-e7004d727432",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(440, 693, 63.49206349206349)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Criteria 12: Variant is a deletion\n",
    "\n",
    "var_list['is_deletion'] = SSC_dnSV_scored[SSC_dnSV_scored.svtype == 'DEL']['index'].values\n",
    "\n",
    "len(var_list['is_deletion']), len(variant_criteria), len(var_list['is_deletion'])/len(variant_criteria)*100"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "d6e62afb-1432-4670-a854-aa6a4fada5f0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n",
      "/tmp/ipykernel_47747/1168745136.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  variant_criteria[criteria_] = 0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>disruptive</th>\n",
       "      <th>not_ASD_gene</th>\n",
       "      <th>no_causal_var</th>\n",
       "      <th>good_pred</th>\n",
       "      <th>no_sib_var</th>\n",
       "      <th>disrupt_CREint</th>\n",
       "      <th>not_expr_gene</th>\n",
       "      <th>not_ExN_RE</th>\n",
       "      <th>near_ASD_gene</th>\n",
       "      <th>change_on_CREint</th>\n",
       "      <th>on_CTCF</th>\n",
       "      <th>is_deletion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>688</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>689</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>690</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>691</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>692</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>693 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     index  disruptive  not_ASD_gene  no_causal_var  good_pred  no_sib_var  \\\n",
       "0        0           0             1              1          0           0   \n",
       "1        1           1             0              0          1           1   \n",
       "2        2           0             1              1          1           1   \n",
       "3        3           0             1              1          1           1   \n",
       "4        4           0             1              1          1           1   \n",
       "..     ...         ...           ...            ...        ...         ...   \n",
       "688    688           0             1              1          0           0   \n",
       "689    689           0             1              1          0           0   \n",
       "690    690           1             1              1          0           1   \n",
       "691    691           0             1              1          0           1   \n",
       "692    692           0             1              1          1           1   \n",
       "\n",
       "     disrupt_CREint  not_expr_gene  not_ExN_RE  near_ASD_gene  \\\n",
       "0                 1              1           0              0   \n",
       "1                 1              0           0              1   \n",
       "2                 1              1           1              1   \n",
       "3                 1              1           0              1   \n",
       "4                 1              1           0              0   \n",
       "..              ...            ...         ...            ...   \n",
       "688               0              1           1              0   \n",
       "689               0              1           0              1   \n",
       "690               0              1           0              0   \n",
       "691               0              0           0              0   \n",
       "692               0              1           0              1   \n",
       "\n",
       "     change_on_CREint  on_CTCF  is_deletion  \n",
       "0                   1        0            1  \n",
       "1                   1        0            1  \n",
       "2                   1        0            1  \n",
       "3                   1        0            1  \n",
       "4                   1        0            1  \n",
       "..                ...      ...          ...  \n",
       "688                 0        0            0  \n",
       "689                 0        1            0  \n",
       "690                 0        0            0  \n",
       "691                 0        0            0  \n",
       "692                 0        0            0  \n",
       "\n",
       "[693 rows x 13 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add vars that corresponds to each criteria to df\n",
    "\n",
    "for criteria_ in criteria:\n",
    "    variant_criteria[criteria_] = 0\n",
    "    variant_criteria.loc[[x in var_list[criteria_] for x in variant_criteria['index']], criteria_] = 1\n",
    "\n",
    "variant_criteria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "98c670b9-90e2-4614-81a8-e59463b310a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "variant_criteria.to_csv(f'{repo_dir}variant_prioritization/variant_criteria_met', sep = '\\t', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
